core/src/main/java/org/jivesoftware/sparkimpl/profile/AvatarPanel.java

```diff
@@ -214,8 +209,16 @@ public void finished() {
                     return;
                 }
                 */
-                setAvatar(new ImageIcon(avatarImage));
-                avatarFile = selectedFile;
+                //convert BufferedImage to bytes
+                try {
+                    ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
You're absolutely right, a ``ByteArrayOutputStream`` does not need to be closed, but many other streams do. It is good practice to simply close all streams that you've created yourself.

As you said, you can do it in the ``finally`` block, but I'd invite you to experiment with the ``try-with-resources`` feature of Java instead. This was introduced in Java 7, and is a slightly more elegant solution.